dandi/files/zarr.py

```diff
@@ -450,6 +450,7 @@ def mkzarr() -> str:
                                 f.cancel()
                             executor.shutdown()
                             client.delete(f"/zarr/{zarr_id}/upload/")
+                            ZARR_UPLOADS_IN_PROGRESS.discard(upload_data)
```

Comment:
uff, took me awhile to locate this `ZARR_UPLOADS_IN_PROGRESS` which is apparently not a local to the function, but  bound at module level below at some line 576, uff . With that it is a global structure, it makes sense.